### PR TITLE
New rejection emails

### PR DESCRIPTION
**Pull request by JR-G**
## Context

We need to ensure the emails sent to candidates still make sense regarding continuous applications.

Currently we have 14 rejection emails, some for API users and some for Manage users. But they are all very similar and trying to tell candidates too many things at once.

There will now be only two rejection emails:

One triggered in Manage

The other triggered for API users

When a candidate receives a Rejection they will receive 1 email per rejected application.

Previously we had a number or emails for rejections depending on what states the candidate’s other applications were in.

Therefore there are some existing emails that will no longer be needed. These must be disabled (and removed later)

## Changes proposed in this pull request

|New rejection email|
|---|
|image
image|

## Guidance to review

Refer to the word doc attached to the trello ticket for content.

You can preview the email at `apply-review-XXXX.test.teacherservices.cloud/rails/mailers/candidate_mailer/application_rejected`

## Link to Trello card

https://trello.com/c/rHKGQeEf/462-ca-emails-design-and-implement-new-rejection-emails

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Review by vassyz — APPROVED** [review]

File: spec/mailers/candidate_mailer_spec.rb [hunk 1]
```diff
@@ -70,6 +70,27 @@
...
+      build_stubbed(:application_form,
+                    candidate:,
+                    application_choices: [application_choice])
+    }
+    let(:email) { mailer.application_rejected(application_choice) }
+
+    before { allow(EmailLogInterceptor).to receive(:generate_reference).and_return('fake-ref-123') }
+
+    context 'when the candidate receives a rejection and continuous applications', :continuous_applications do
+      it_behaves_like(
```
Comment (avinhurry) [inline spec/mailers/candidate_mailer_spec.rb:85]: Do we need `:continuous_applications` in this test?

File: spec/mailers/candidate_mailer_spec.rb [hunk 2]
```diff
@@ -70,6 +70,27 @@
...
+        'a mail with subject and content',
+        I18n.t!('candidate_mailer.application_rejected.subject'),
```
Comment (avinhurry) [inline spec/mailers/candidate_mailer_spec.rb:87]: How does `I18n.t!` differ to `I18n.t`?

File: spec/services/send_candidate_rejection_email_spec.rb [hunk 3]
```diff
@@ -14,7 +14,7 @@
     let(:application_choice_offer_deferred) { create(:application_choice, status: :offer_deferred, application_form:) }
     let(:mail) { instance_double(ActionMailer::MessageDelivery, deliver_later: true) }
 
-    context 'when an application choice is rejected' do
+    context 'when an application choice is rejected and it is not continous applications', continuous_applications: false do
```
Comment (avinhurry) [inline spec/services/send_candidate_rejection_email_spec.rb:17]: I feel the `continuous_applications: false` is enough of a description

File: spec/system/candidate_interface/continuous_applications/rejections/candidate_receives_rejection_email_spec.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+RSpec.feature 'Receives rejection email', :continuous_applications do
+  include CandidateHelper
+
+  scenario 'Receives rejection email' do
+    when_i_have_submitted_an_application
```
Comment (avinhurry) [inline spec/system/candidate_interface/continuous_applications/rejections/candidate_receives_rejection_email_spec.rb:7]: Are we not too fussy about using the cucumber style syntax on Apply?

**Review by avinhurry — APPROVED** [review]
Looks good. Defo no blockers, mainly curiosity.

**Review by MaeveRoseveare — APPROVED** [review]

**Pull request merged**